src/ecdh.rs

```diff
@@ -201,38 +188,6 @@ mod tests {
         assert_eq!(sec1, sec2);
         assert_ne!(sec_odd, sec2);
     }
-
-    #[test]
-    fn ecdh_with_hash_callback() {
-        let s = Secp256k1::signing_only();
-        let (sk1, pk1) = s.generate_keypair(&mut thread_rng());
-        let expect_result: [u8; 64] = [123; 64];
-        let mut x_out = [0u8; 32];
-        let mut y_out = [0u8; 32];
-        let result = SharedSecret::new_with_hash(&pk1, &sk1, |x, y| {
-            x_out = x;
-            y_out = y;
-            expect_result.into()
-        });
-        assert_eq!(&expect_result[..], &result[..]);
-        assert_ne!(x_out, [0u8; 32]);
-        assert_ne!(y_out, [0u8; 32]);
-    }
-
-    #[test]
-    fn test_c_callback() {
-        let x = [5u8; 32];
-        let y = [7u8; 32];
-        let mut output = [0u8; 64];
-        let res = unsafe { super::c_callback(output.as_mut_ptr(), x.as_ptr(), y.as_ptr(), ::ptr::null_mut()) };
-        assert_eq!(res, 1);
-        let mut new_x = [0u8; 32];
-        let mut new_y = [0u8; 32];
-        new_x.copy_from_slice(&output[..32]);
-        new_y.copy_from_slice(&output[32..]);
-        assert_eq!(x, new_x);
-        assert_eq!(y, new_y);
-    }
```

Comment:
Why are these tests removed?

Reply:
I've reintroduced both tests along with the unsafe `c_callback` function. However for some reason `x_out` is NOT null thus the `ecdh_with_hash_callback` test fails.

Reply:
Do we have any idea why the test is failing here?